src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -423,5 +423,76 @@ private static string NormalizeDateTimeCollectionItems(string bracketLiteralText
 
             return "[" + String.Join(",", items) + "]";
         }
+
+        private static bool IsCollectionContentEmptyOrSpaces(string bracketLiteralText)
+        {
+            if (string.IsNullOrWhiteSpace(bracketLiteralText) || bracketLiteralText.Length < 2)
+            {
+                return true;
+            }
+
+            string content = bracketLiteralText[1..^1].Trim();
+
+            if (string.IsNullOrWhiteSpace(content))
+            {
+                return true;
+            }
+
+            bool isEmptyOrHasOnlySpaces = true;
+            bool isCharinsideQuotes = false;
+            char quoteChar = '\0';
+            Span<char> buffer = stackalloc char[content.Length];

```

Comment:
Do we know the bounds for content? If it is user allocated how will this work to ensure that we will not incur a Stack overflow exception.

Reply:
The bounds for content is the length of the content.. The user cannot set if. We get it from whatever the user passes in brackets..

Reply:
Or we throw an exception if an upper bound is exceeded.

Reply:
Updated this to use an ArrayPool instead.